From b1dd8738ea090976a0b6ed598be658515d2cadbc Mon Sep 17 00:00:00 2001
From: faux123 <reioux@gmail.com>
Date: Fri, 11 Jan 2013 21:28:43 -0800
Subject: [PATCH 093/507] intelli_plug: increase cores on persistence

also replace hardcoded values with Macros for ease of updating later

Signed-off-by: faux123 <reioux@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 arch/arm/mach-msm/intelli_plug.c |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-msm/intelli_plug.c b/arch/arm/mach-msm/intelli_plug.c
index 4a5a166..9374ecb 100644
--- a/arch/arm/mach-msm/intelli_plug.c
+++ b/arch/arm/mach-msm/intelli_plug.c
@@ -26,6 +26,10 @@
 #define DEF_SAMPLING_RATE		(50000)
 #define DEF_SAMPLING_MS			(50)
 
+#define DUAL_CORE_PERSISTENCE		50
+#define TRI_CORE_PERSISTENCE		40
+#define QUAD_CORE_PERSISTENCE		30
+
 static DEFINE_MUTEX(intelli_plug_mutex);
 
 struct delayed_work intelli_plug_work;
@@ -116,7 +120,7 @@ static void intelli_plug_work_fn(struct work_struct *work)
 					//pr_info("case 1: %u\n", persist_count);
 					break;
 				case 2:
-					persist_count = 27;
+					persist_count = DUAL_CORE_PERSISTENCE;
 					if (num_online_cpus() == 1)
 						cpu_up(1);
 					else {
@@ -126,7 +130,7 @@ static void intelli_plug_work_fn(struct work_struct *work)
 					//pr_info("case 2: %u\n", persist_count);
 					break;
 				case 3:
-					persist_count = 21;
+					persist_count = TRI_CORE_PERSISTENCE;
 					if (num_online_cpus() == 2)
 						cpu_up(2);
 					else
@@ -134,7 +138,7 @@ static void intelli_plug_work_fn(struct work_struct *work)
 					//pr_info("case 3: %u\n", persist_count);
 					break;
 				case 4:
-					persist_count = 15;
+					persist_count = QUAD_CORE_PERSISTENCE;
 					if (num_online_cpus() == 3)
 						cpu_up(3);
 					//pr_info("case 4: %u\n", persist_count);
@@ -176,7 +180,7 @@ static void intelli_plug_late_resume(struct early_suspend *handler)
 
 	mutex_lock(&intelli_plug_mutex);
 	/* keep cores awake long enough for faster wake up */
-	persist_count = 27;
+	persist_count = DUAL_CORE_PERSISTENCE;
 	suspended = false;
 	mutex_unlock(&intelli_plug_mutex);
 
-- 
1.7.9.5

